### PR TITLE
refactor user-context

### PR DESCRIPTION
**Pull request by hsnice16**
In this PR, user-context has been removed with userSlice. 

Please review this PR and give your feedback.

### CODE REVIEW

**Review by mikesch803 — APPROVED** [review]
nice that you are refactoring your code to redux, nothing to suggest, good to merge

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@emotion/react": "^11.9.0",
     "@emotion/styled": "^11.8.1",
     "@mui/material": "^5.6.2",
+    "@reduxjs/toolkit": "^1.8.1",
     "axios": "^0.27.2",
     "classnames": "^2.3.1",
```
Comment (Dev-AkshatGupta) [inline package.json:11]: What is this packcage for?

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "@emotion/react": "^11.9.0",
     "@emotion/styled": "^11.8.1",
     "@mui/material": "^5.6.2",
```
Comment (Dev-AkshatGupta) [inline package.json:8]: great that you are using material design

File: src/components/left-side-navbar/LeftSideNavbar.jsx [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import { useState } from "react";
 import classNames from "classnames";
+import { useProfile } from "context";
 import { LeftSideNavbarData } from "data";
-import { useProfile, useUser } from "context";
+import { useSelector } from "react-redux";
 import { EllipsisHorizontalIcon } from "assets";
 import styles from "./LeftSideNavbar.module.css";
 import { ROUTE_HOME, ROUTE_PROFILE } from "utils";
```
Comment (Dev-AkshatGupta) [inline src/components/left-side-navbar/LeftSideNavbar.jsx:8]: What are these variables exactly for

**Review by Dev-AkshatGupta — APPROVED** [review]
Everything seems really good to me and you great that you are using the redux rtk queries
You can go ahead and merge it

File: src/components/left-side-navbar/LeftSideNavbar.jsx (hunk 3 above)
Comment (hsnice16) [reply src/components/left-side-navbar/LeftSideNavbar.jsx:8]: These constants are for maintaining the route path.

File: package.json (hunk 1 above)
Comment (hsnice16) [reply package.json:11]: Suppose you have many classes, and that is for a single element only, and you are also adding some classes via some condition.
So, generally, we use template literal for that, but I think the readability of the code decreases as the number of conditional classes gets increases.
So, using this package we can keep the readability at a good level. 

If you want to read more, https://www.npmjs.com/package/classnames

**Comment by hsnice16** [comment]
Thank you for reviewing,

**Pull request merged**